### ISSUE TITLE
Exception in thread "JavaFX Application Thread"

### ISSUE THREAD

**Issue by klabarge** [qzind/tray#793]
I get this error when sending an invalid html print config (for example, a 4x6in label with 2in margins)

`size:{width: 4, height: 6}`
`margins: 2`

The print error is probably sane, however QZ Tray never recovers. I am unable to print using QZ Tray until I force close QZ Tray.

**OS:** macOS 11.2.2 (20D80)
**Java:** AdoptOpenJDK (build 11.0.10+9)

```
Exception in thread "JavaFX Application Thread" java.lang.IllegalArgumentException: 0 or negative value argument
	at java.desktop/javax.print.attribute.standard.MediaPrintableArea.<init>(MediaPrintableArea.java:127)
	at com.sun.prism.j2d.print.J2DPrinterJob.syncPageLayout(J2DPrinterJob.java:691)
	at com.sun.prism.j2d.print.J2DPrinterJob.syncSettingsToAttributes(J2DPrinterJob.java:596)
	at com.sun.prism.j2d.print.J2DPrinterJob.print(J2DPrinterJob.java:822)
	at javafx.print.PrinterJob.printPage(PrinterJob.java:399)
	at javafx.print.PrinterJob.printPage(PrinterJob.java:415)
	at qz.printer.action.WebApp.lambda$print$5(Unknown Source)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
```

Here you can see 8 minutes go by from sending a print job and having to close QZ Tray.

```
[DEBUG] 2021-04-29 19:46:08,073 @ qz.utils.PrintingUtilities:?
	Using qz.printer.action.PrintHTML to print
[DEBUG] 2021-04-29 19:46:08,074 @ qz.printer.PrintServiceMatcher:?
	Searching for PrintService matching PDFwriter
[DEBUG] 2021-04-29 19:46:08,199 @ qz.printer.PrintServiceMatcher:?
	Found 6 printers
[DEBUG] 2021-04-29 19:46:08,300 @ qz.printer.PrintServiceMatcher:?
	Found match: PDFwriter
[WARN] 2021-04-29 19:46:08,304 @ qz.printer.PrintOptions:?
	Default resolution for PDFwriter is missing, using fallback: null
[WARN] 2021-04-29 19:46:08,304 @ qz.printer.PrintOptions:?
	Cannot read null as a double for fallbackDensity, using default
[INFO] 2021-04-29 19:54:06,248 @ qz.ws.PrintSocketClient:?
	Connection closed: 1005 - null
[INFO] 2021-04-29 19:54:06,249 @ qz.ws.SocketConnection:?
	Closing all communication channels for localhost
^C
ççççç[DEBUG] 2021-04-29 19:54:21,260 @ qz.printer.action.PrintHTML:?
	Parsed 1 html records
[ERROR] 2021-04-29 19:54:21,262 @ qz.utils.PrintingUtilities:?
	Failed to print
java.awt.print.PrinterException
	at qz.printer.action.PrintHTML.print(Unknown Source)
	at qz.utils.PrintingUtilities.processPrintRequest(Unknown Source)
	at qz.ws.PrintSocketClient.processMessage(Unknown Source)
	at qz.ws.PrintSocketClient.onMessage(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.jetty.websocket.common.events.annotated.CallableMethod.call(CallableMethod.java:70)
	at org.eclipse.jetty.websocket.common.events.annotated.OptionalSessionCallableMethod.call(OptionalSessionCallableMethod.java:68)
	at org.eclipse.jetty.websocket.common.events.JettyAnnotatedEventDriver$2.run(JettyAnnotatedEventDriver.java:227)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:914)
	at java.base/java.lang.Thread.run(Thread.java:834)
[WARN] 2021-04-29 19:54:21,266 @ org.eclipse.jetty.websocket.common.extensions.compress.CompressExtension:455
	
java.nio.channels.ClosedChannelException
	at org.eclipse.jetty.websocket.common.io.FrameFlusher.enqueue(FrameFlusher.java:109)
	at org.eclipse.jetty.websocket.common.io.AbstractWebSocketConnection.outgoingFrame(AbstractWebSocketConnection.java:575)
	at org.eclipse.jetty.websocket.common.extensions.AbstractExtension.nextOutgoingFrame(AbstractExtension.java:155)
	at org.eclipse.jetty.websocket.common.extensions.compress.CompressExtension.access$1100(CompressExtension.java:44)
	at org.eclipse.jetty.websocket.common.extensions.compress.CompressExtension$Flusher.compress(CompressExtension.java:591)
	at org.eclipse.jetty.websocket.common.extensions.compress.CompressExtension$Flusher.deflate(CompressExtension.java:488)
	at org.eclipse.jetty.websocket.common.extensions.compress.CompressExtension$Flusher.process(CompressExtension.java:468)
	at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:241)
	at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:223)
	at org.eclipse.jetty.websocket.common.extensions.compress.CompressExtension.outgoingFrame(CompressExtension.java:244)
	at org.eclipse.jetty.websocket.common.extensions.ExtensionStack$Flusher.process(ExtensionStack.java:400)
	at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:241)
	at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:223)
	at org.eclipse.jetty.websocket.common.extensions.ExtensionStack.outgoingFrame(ExtensionStack.java:277)
	at org.eclipse.jetty.websocket.common.WebSocketRemoteEndpoint.uncheckedSendFrame(WebSocketRemoteEndpoint.java:306)
	at org.eclipse.jetty.websocket.common.WebSocketRemoteEndpoint.blockingWrite(WebSocketRemoteEndpoint.java:106)
	at org.eclipse.jetty.websocket.common.WebSocketRemoteEndpoint.sendString(WebSocketRemoteEndpoint.java:384)
	at qz.ws.PrintSocketClient.send(Unknown Source)
	at qz.ws.PrintSocketClient.sendError(Unknown Source)
	at qz.ws.PrintSocketClient.sendError(Unknown Source)
	at qz.utils.PrintingUtilities.processPrintRequest(Unknown Source)
	at qz.ws.PrintSocketClient.processMessage(Unknown Source)
	at qz.ws.PrintSocketClient.onMessage(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.jetty.websocket.common.events.annotated.CallableMethod.call(CallableMethod.java:70)
	at org.eclipse.jetty.websocket.common.events.annotated.OptionalSessionCallableMethod.call(OptionalSessionCallableMethod.java:68)
	at org.eclipse.jetty.websocket.common.events.JettyAnnotatedEventDriver$2.run(JettyAnnotatedEventDriver.java:227)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:914)
	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 2021-04-29 19:54:21,267 @ qz.ws.PrintSocketClient:?
	Could not send message
```

**Comment by tresf** [comment]
Thanks, I think we can guard against this.  @bberenz?

**Comment by tresf** [comment]
Closed via #793
